cwe_checker_rs/src/checkers/cwe_78/context/mod.rs

```diff
@@ -0,0 +1,563 @@
+            new_state.set_pi_def_map(self.create_pi_def_map(blk_start_node));
+        }
+
+        match &def.term {
+            Def::Assign { var, value: input }
+            | Def::Load {
+                var,
+                address: input,
+            } => new_state = self.handle_assign_and_load(new_state, def, var, input),
+            Def::Store { address, value } => {
+                new_state = self.handle_store(new_state, def, address, value)
+            }
+        }
+
+        Some(new_state)
+    }
+
+    /// Either returns a copy of the input state when there is no conditional
+    /// Or merges both incoming states from the branch and conditional branch
+    fn update_jumpsite(
+        &self,
+        state_after_jump: &State,
+        _jump: &Term<Jmp>,
+        _untaken_conditional: Option<&Term<Jmp>>,
+        _jumpsite: &Term<Blk>,
+    ) -> Option<State> {
+        Some(state_after_jump.clone())
+    }
+
+    fn update_callsite(
+        &self,
+        target_state: Option<&State>,
+        return_state: Option<&State>,
+        _call: &Term<Jmp>,
+        _return_: &Term<Jmp>,
+    ) -> Option<State> {
+        if let Some(return_) = return_state {
+            // If there is both a return and target state, first remove all non parameter
+            // taints from the target state and merge in all callee saved register taints from
+            // the return state. The new state will contain possible parameter taints, possible callee saved taints and
+            // possible memory taints
+            if let Some(target) = target_state {
+                let mut new_state = target.clone();
+                new_state.remove_non_parameter_taints_for_generic_function(self.project);
+                if let Some(calling_conv) = self.project.get_standard_calling_convention() {
+                    let callee_saved_registers: HashSet<String> =
+                        calling_conv.callee_saved_register.iter().cloned().collect();
+                    for (variable, taint) in return_.get_register_taints() {
+                        if callee_saved_registers.get(&variable.name).is_some() {
+                            new_state.set_register_taint(variable, *taint);
+                        }
+                    }
+
+                    return Some(new_state);
+                }
+            }
+            // If there is no target state, try to get the standard calling convention of the called function
+            // and remove all non callee saved registers taints from the state at the return side
+            let mut new_state = return_.clone();
+            if let Some(calling_conv) = self.project.get_standard_calling_convention() {
+                new_state.remove_non_callee_saved_taint(calling_conv);
+            }
+
+            return Some(new_state);
+        }
+
+        // If there no return state, there will also be no target state and therefore None is returned
+        None
```

Comment:
What if the taint source/system call was inside the callee? Then there would be a target_state without a return_state.